### PR TITLE
[CINN]Apply input dynamic dim specification

### PR DESCRIPTION
**Pull request by Hongqing-work**
### PR Category

CINN

### PR Types

New features

### Description

Pcard-67164
This PR support input dynamic dim specification.
More information about this can be found in [PR67384](https://github.com/PaddlePaddle/Paddle/pull/67384)
This json format example:
```json
{
    "batch_size": {
        "input_bind": [
            ["input_ids", 0],
            ["attention_mask", 0],
            ["position_ids", 0]
        ],
        "min": 2
    },
    "sequence_length": {
        "input_bind": [
            ["input_ids", 1],
            ["attention_mask", 1],
            ["position_ids", 1]
        ],
        "max": 128
    }
}
```

### CODE REVIEW

File: paddle/common/flags.cc [hunk 1]
```diff
@@ -1104,6 +1104,32 @@ PHI_DEFINE_EXPORTED_string(cinn_subgraph_graphviz_dir,
...
+ * CINN related FLAG
+ * Name: FLAGS_cinn_specify_input_dynamic_dim
+ * Since Version: develop
+ * Value Range: bool, default=false
+ * Example: FLAGS_cinn_specify_input_dynamic_dim=true will use file set by
+ * FLAGS_cinn_input_dynamic_dim_spec_file to specify input dynamic dimention.
+ */
+PHI_DEFINE_EXPORTED_bool(cinn_specify_input_dynamic_dim,
+                         false,
+                         "Whether to specify input dynamic dimention.");
```
Comment (zyfncg) [inline paddle/common/flags.cc:1117]: 这个flag是不可以不用添加？

**Review by zyfncg — APPROVED** [review]

**Review by gongshaotian — APPROVED** [review]

File: paddle/common/flags.cc (hunk 1 above)
Comment (Hongqing-work) [reply paddle/common/flags.cc:1117]: 之后考虑给path指定默认值，所以这里需要一个直接表明是否开启的开关

File: paddle/common/flags.cc [hunk 2]
```diff
@@ -1104,6 +1104,32 @@ PHI_DEFINE_EXPORTED_string(cinn_subgraph_graphviz_dir,
...
+
+/*
+ * CINN related FLAG
+ * Name: FLAGS_cinn_input_dynamic_dim_spec_file
+ * Since Version: develop
```
Comment (Aurelius84) [inline paddle/common/flags.cc:1122]:
```suggestion
 * Since Version: 3.0 beta2
```

**Review by Aurelius84 — APPROVED** [review]
LGTM for flags

**Review by yuanlehome — APPROVED** [review]

File: paddle/common/flags.cc (hunk 2 above)
Comment (Hongqing-work) [reply paddle/common/flags.cc:1122]: 收到，下一个PR顺带修改
